source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -231,6 +237,92 @@ def image(
     return images.clone()
 
 
+class image_features(ManagerTermBase):
+    """Extracted image features with a frozen encoder from images of a specific datatype from the camera sensor.
+
+    Calls :meth:`image` to get the images, then performs inference. On initialization,
+    for a model zoo different from the default, define model_zoo_cfg: A dictionary with string keys and callable values.
+    Should include "model", (mapped to a callable with no arguments to return the model), "preprocess" (mapped to
+    a callable which consumes the images and returns the preprocessed images),
+    and "inference" (mapped to a callable that provided the model, and the preproccessed images, returns the features.)
+    """
+
+    def __init__(
+        self,
+        cfg: ObservationTermCfg,
+        env: ManagerBasedEnv,
+        model_zoo_cfg: dict | None = None,
+        initialize_all: bool = False,
+        device: str = "cuda:0",
```

Comment:
Why is the device needed here?

Reply:
Vision transformer models like Theia take up a lot of GPU mem, especially for inference, reducing the number of possible environments on that GPU. Allows for EZ offload of this computation to another GPU to increase possible env count when using image features

Reply:
For example, could have main strong gpu run sim, and a secondary  weaker one inference the model